### PR TITLE
task: generate model from code asset

### PR DESCRIPTION
**Pull request by blanchco**
# Description

* generate model from code asset in model from code operator

https://github.com/DARPA-ASKEM/terarium/assets/33158416/5491c718-16ad-49c6-a8aa-3d942b2eeb84

### CODE REVIEW

File: packages/client/hmi-client/src/workflow/ops/model-from-code/tera-model-from-code.vue [hunk 1]
```diff
@@ -183,14 +199,8 @@ onUnmounted(() => {
...
-	const contextName =
-		selectedModelFramework.value === ModelFramework.Decapodes ? 'decapodes_creation' : null;
-	const languageName =
-		selectedProgrammingLanguage.value === ProgrammingLanguage.Julia ? 'julia-1.9' : null;
-	if (contextName === null || languageName === null) {
-		logger.warn("Can't work with current language. Do nothing.");
-		return null;
-	}
+	const contextName = 'decapodes_creation';
+	const languageName = 'julia-1.9';
```
Comment (blanchco) [inline packages/client/hmi-client/src/workflow/ops/model-from-code/tera-model-from-code.vue:203]: statically settings these for now - We will only run the kernel when the decapodes options is selected anyways

Comment (YohannParis) [reply packages/client/hmi-client/src/workflow/ops/model-from-code/tera-model-from-code.vue:203]: I agree with you, but why not leaving that nice test there?

File: packages/client/hmi-client/src/workflow/ops/model-from-code/tera-model-from-code.vue [hunk 2]
```diff
@@ -235,22 +256,28 @@ function getModel() {
...
+				model_version: 'v1.0',
+				schema: 'modelreps.io/DecaExpr',
+				schema_name: 'DecaExpr'
+			}
+		};
+
+		modelValid.value = false;
+
+		kernelManager
+			.sendMessage('save_amr_request', messageContent)
```
Comment (YohannParis) [inline packages/client/hmi-client/src/workflow/ops/model-from-code/tera-model-from-code.vue:278]: Why is the AMR saved via the kernel?

Comment (blanchco) [reply packages/client/hmi-client/src/workflow/ops/model-from-code/tera-model-from-code.vue:278]: saving for petrinet will happen in another PR - This save message is just for the decapodes flow not the petrinet

Comment (YohannParis) [reply packages/client/hmi-client/src/workflow/ops/model-from-code/tera-model-from-code.vue:278]: I agree, still why is it happening within the Kernel?

Comment (blanchco) [reply packages/client/hmi-client/src/workflow/ops/model-from-code/tera-model-from-code.vue:280]: I couldnt tell you why its happening that way with the kernel, maybe something for @mwdchang to answer?  Are you saying you would want get the kernel response and then save it with a REST call?

File: packages/client/hmi-client/src/workflow/ops/model-from-code/tera-model-from-code.vue (hunk 1 above)
Comment (blanchco) [reply packages/client/hmi-client/src/workflow/ops/model-from-code/tera-model-from-code.vue]: Youre right - I've added the tests for contextName and languageName but removed the warning and the early null return.  We still want the context to be setup when the component is mounted and returning null wouldn't let that happen

File: packages/client/hmi-client/src/workflow/ops/model-from-code/tera-model-from-code.vue (hunk 2 above)
Comment (blanchco) [reply packages/client/hmi-client/src/workflow/ops/model-from-code/tera-model-from-code.vue:280]: in any case I feel that change would be better in a followup PR as well

File: packages/client/hmi-client/src/workflow/ops/model-from-code/tera-model-from-code.vue [hunk 3]
```diff
@@ -144,28 +144,41 @@ enum ModelFramework {
 const isProcessing = ref(false);
 
 const editor = ref<VAceEditorInstance['_editor'] | null>(null);
-const modelName = ref('');
```
Comment (blanchco) [inline packages/client/hmi-client/src/workflow/ops/model-from-code/tera-model-from-code.vue:147]: removing the modelname for now - I will handle this in a later PR.  But the flow to name a model is a little confusing at the moment

**Review by YohannParis — APPROVED** [review]

**Pull request merged**